packages/vscode-extension/src/devices/IosSimulatorDevice.ts

```diff
@@ -706,6 +709,87 @@ export class IosSimulatorDevice extends DeviceBase {
     await exec("xcrun", args);
     return { canSafelyRemove: true };
   }
+
+  protected async runMaestroTest(fileNames: string[]) {
+    this.maestroLogsOutputChannel.show(true);
+    this.maestroLogsOutputChannel.appendLine("");
+
+    if (fileNames.length === 1) {
+      const fileName = fileNames[0];
+      const isFile = fs.lstatSync(fileName).isFile();
```

Comment:
Since we're in `async` context anyway, it would be better to use the `fs.promises` version of the API instead, to avoid blocking JS execution while this runs.